src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -71,15 +98,51 @@ public void close() throws Exception {
     // Mostly for JUnit tests
     motor.close();
     motor2.close();
-    
   }
 
-  @Override
-  public void periodic() {
-    intakeMode.setString((isConeMode) ? "Cone Mode" : "Cube Mode");
-    // This method will be called once per scheduler run
+  public double getConeDist() {
+    return timeOfFlightSensor.getRawDistance0().classDistance;
   }
 
+  public double getCubeDist() {
+    return timeOfFlightSensor.getRawDistance1().classDistance;
+  }
+
+  public void changeFlightState() {
+    double coneDistance = getConeDist();
+    double cubeDistance = getCubeDist();
+
+    boolean coneSensorUp = coneDistance != -1;
+    boolean cubeSensorUp = cubeDistance != -1;
+
+    coneDist.log(coneDistance);
+    cubeDist.log(cubeDistance);
+    hasCone.log(coneDistance <= coneActivationThreshold);
+    hasCube.log(cubeDistance <= cubeActivationThreshold);
+    isConeSensor.log(coneSensorUp);
+    isCubeSensor.log(cubeSensorUp);
+
+    // Only change state if both sensors are up, otherwise stay in last state
+    if (coneDistance <= coneActivationThreshold && coneSensorUp) {
+      tofState = FlightStates.CONE;
+    } 
+    else if (cubeDistance <= cubeActivationThreshold && cubeSensorUp) {
+      tofState = FlightStates.CUBE;
+    } 
+    else {
+      tofState = FlightStates.IDLE;
+    }
```

Comment:
If the sensors are down, this will bring it back to IDLE instead of keeping the last state

Reply:
Changed to only switch to idle if at least 1 sensor is up